application-task-manager-default/src/main/java/com/xwiki/taskmanager/internal/macro/TaskMacro.java

```diff
@@ -94,19 +106,33 @@ public List<Block> execute(TaskMacroParameters parameters, String content, Macro
         this.ssx.use(DateMacro.SKIN_RESOURCES_DOCUMENT_REFERENCE);
         this.jsx.use(DateMacro.SKIN_RESOURCES_DOCUMENT_REFERENCE);
 
-        Block ret = null;
+        List<Block> contentBlocks = new ArrayList<>();
 
-        if (content == null) {
-            return Collections.emptyList();
-        }
+        Task task = taskManager.getTaskByReference(parameters.getReference());
 
-        Map<String, String> blockParameters = new HashMap<>();
+        if (content != null && !content.trim().isEmpty()) {
+            try {
+                List<Block> macroContent =
+                    taskBlockProcessor.getTaskContentXDOM(context.getCurrentMacroBlock(), context.getSyntax())
+                        .getChildren();
 
-        List<Block> macroContent = contentParser.parse(content, context, false, context.isInline())
-            .getChildren();
+                contentBlocks =
+                    Collections.singletonList(new MetaDataBlock(macroContent, this.getNonGeneratedContentMetaData()));
+            } catch (TaskException e) {
+                throw new MacroExecutionException("Failed to get the content of the task.", e);
+            }
+        } else if (parameters.getReference() != null) {
+            contentBlocks = getContentFromTaskPage(task, parameters.getReference());
```

Comment:
We talked about not supporting this and instead have another macro for task display (e.g. ``{{tasks ids="..."/}}``).